website/blog/2023-05-05-making-your-project-multilingual.md

```diff
@@ -0,0 +1,85 @@
+---
+title: Making your project multilingual
+description: Examples of using I18n in a project
+slug: making-your-project-multilingual
+authors: obradovic
+date: 2023-05-05
+tags: [eightshift, boilerplate, i18n, multilingual]
+hide_table_of_contents: false
+---
+
+Automatic website translation with tools like Google Translate has become quite decent, but it is still a much better approach to add multiple languages to your project and maintain the translations for those languages yourself.
+<!--truncate-->
+
+## Translating the strings in PHP
+It is good practice to always wrap your hardcoded strings in one of the functions used for I18n (internationalization). Even if your website may initially be only in one language, it will be much easier if you have to add multilingual support later.
+
+If you've already worked on a multilingual project, you most likely came across `__()` and `_e()` functions. The main difference between the `__()` and `_e()` is that `__()` only returns the value, while `_e()` also echoes it. Both of these functions accept two arguments, the first one is your string, and the second one is your textdomain.
+
+> Textdomain is usually your project name written in kebab-case.
+
+WordPress functions like `__()` and `_e()` do the job quite well, but it is much better to use the variants of these functions that also escape the output. These are `esc_html__()` and `esc_html_e()`. There are also a few more functions for I18n you can use, but to keep it simple, we'll just mention these two for now.
+
+Here is an example of using one of these functions:
+```php
+<?php echo esc_html__('Contact', 'project-name'); ?>
+```
```

Comment:
Maybe also mention the [`_n`](https://developer.wordpress.org/reference/functions/_n/) function?

Reply:
I mentioned it now in **Translating with Poedit** section